services/src/main/java/io/grpc/services/BinaryLogProviderImpl.java

```diff
@@ -52,14 +52,22 @@ public BinaryLogProviderImpl() {
   @Nullable
   @Override
   public ServerInterceptor getServerInterceptor(String fullMethodName) {
-    return factory.getLog(fullMethodName).getServerInterceptor(getServerCallId());
+    BinaryLog log = factory.getLog(fullMethodName);
```

Comment:
lines 55-58 and 66-69 are the same and could effectively be pulled out to a helper method so the code is shared.